include/boost/math/tools/polynomial.hpp

```diff
@@ -134,7 +236,11 @@ class polynomial
 
    // access:
    size_type size()const { return m_data.size(); }
-   size_type degree()const { return m_data.size() - 1; }
+   size_type degree()const
+   { 
+       BOOST_ASSERT(size() != 0);
```

Comment:
If the size is zero, then the order is zero (with the order zero coefficient zero), so I think I'd prefer:

if(empty()) return 0;

rather than the assert.